Feature/FA-21-Reset-Password

Ammar720 opened the pull request:
## 📝 Description

This pull request introduces the **Reset Password** feature in the e-commerce flower app. It includes updates across the data, domain, and presentation layers to support secure password updating, input validation, and full API-UI integration using MVI architecture.



## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format:Feature/FA-21-Reset-Password.
- [x] The branch name follows the required format: feature/fa-21-reset-password.

### 📂 Code Quality
- [x] All filenames are in snake_case.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with dart format .

### 🛠 CI/CD Validation
- [x] Linting passes with dart analyze.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [ ] Unit tests are included for all new features and bug fixes.
- [ ] flutter test --coverage has been run and the results reviewed.
- [ ] The project builds successfully without errors.

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: FA-21

### 📸 Screenshots (if applicable):

<p float="left">
  <img src="https://github.com/user-attachments/assets/2ccc94d4-6a0f-484d-b8bd-f481948c55d4" width="45%" />

  <img src="https://github.com/user-attachments/assets/dc27cedf-e64d-43f7-b1d6-69c97a018ad0" width="45%" />
</p>

Commit:
feat(reset_password): implement reset password Data and Domain layers

Commit:
feat(reset_password): implement reset password view model and fix api route

Commit:
feat(reset_password): implement reset password UI and integrate it with the API

Commit:
Merge branch 'developement' into feature/fa-21-reset-password

Commit:
refactor(reset_password):change all files names and related fields from reset password to change password

ali72-20 commented on lib/features/change_password/data/data_source/contract/change_password_data_source.dart:
```diff
@@ -0,0 +1,8 @@
+import 'package:ecommerce_flower_app/features/change_password/data/models/change_password_request_dto/change_password_request_dto.dart';
+import 'package:ecommerce_flower_app/features/change_password/data/models/change_password_response_dto/change_password_response_dto.dart';
+
+abstract class ChangePasswordDataSource {
```
add online data source or local data source to be more clear 
ex: ChangePasssordOnlineDataSource

ali72-20 reviewed (approved):
perfect work! 
a very small things to note but it's still perfect

Commit:
Merge branch 'developement' into feature/fa-21-reset-password

Commit:
refactor(change_password): rename ChangePasswordDataSource to ChangePasswordRemoteDataSource and update references

Commit:
Merge remote-tracking branch 'origin/developement' into feature/fa-21-reset-password

Commit:
refactor(cart): refactor data model serialization for cart and product features

- Updated JSON serialization methods in `add_product_to_cart_request_dto`, `cart`, `cart_items`, `product`, and related DTOs to improve readability and maintainability.
- Changed field names from `Id` to `id` for consistency across models.
- Enhanced formatting in generated code for better clarity.
- Adjusted API client implementations for categories, home, and occasions to streamline request handling.
- Improved handling of null values and type conversions in JSON parsing.
- Updated `product_details_cubit` to use `flutter_bloc` package for state management.
